File: Zipcode Parsing/zipcodes.cs
using System;
using System.Linq;
using System.Collections.Generic;
using System.Text;
using System.IO;
using System.Collections.Specialized;


	public class zipcodes {

	public static List<String> citiesList = new List<String>();
	public static List<String> zipsList = new List<String>();
	public static List<String> statesList = new List<String>();
	public static List<String> commonCityNames = new List<String>();
	public static List<String> latitude = new List<String>();

	public static NameValueCollection nvc = new NameValueCollection();
	public static NameValueCollection commonCity = new NameValueCollection();


	
    public static void populateLists()
    {
        StreamReader input = new StreamReader("cities.txt");
        while (!input.EndOfStream)
        {
            string line = input.ReadLine();
            string[] toks = line.Split('\t');

            foreach (string value in toks)
            {
                citiesList.Add(value);
            }


        }

        input.Close();


        input = new StreamReader("zips.txt");
        while (!input.EndOfStream)
        {
            string line = input.ReadLine();
            string[] toks = line.Split('\t');

            foreach (string value in toks)
            {


                zipsList.Add(value);
            }


        }

        input.Close();


        input = new StreamReader("states.txt");
        while (!input.EndOfStream)
        {
            string line = input.ReadLine();
            string[] toks = line.Split('\t');

            foreach (string value in toks)
            {


                statesList.Add(value);
            }


        }

        input.Close();
    }

	


    public static void CommonCityNames(){


            List<String> printList = new List<String>();
	    List<String> finalList = new List<String>();
	    bool populatedFinalList = false;
		StreamWriter output = new StreamWriter("CommonCityNames.txt");
	 	
	

		StreamReader input = new StreamReader("zipcodes.txt");
			
			while (!input.EndOfStream) {
				string line = input.ReadLine();
				string[] toks = line.Split('\t');
				if(statesList.Contains(toks[4])){
				
					nvc.Add(toks[4], toks[3]);

				}
				
			}
	   string[] values = null;


	  
           foreach (string key in statesList) {


		// Clear the printList
		printList.Clear();
		

                values = nvc.GetValues(key);

                foreach (string value in values)
                {
                      printList.Add(value);

                } 

		if(populatedFinalList == false){

			finalList = printList;

			populatedFinalList = true;
		}

		
		List<String> tmpList = new List<String>();
		tmpList = printList.Intersect(finalList).ToList();

		finalList = tmpList;


	
	} // end iteration
	  
		finalList.Sort();
		foreach (string value in finalList){
                    
		  output.Write(value);
		  output.Write("\n");

               } // end foreach
 	
	    
		

   
		output.Close();
} 


    public static void LatLon(){

		StreamReader input = new StreamReader("zipcodes.txt");
			
			while (!input.EndOfStream) {
				string line = input.ReadLine();
				string[] toks = line.Split('\t');
				
				if(zipsList.Contains(toks[1])){
					
					latitude.Add(toks[6]);
					latitude.Add(toks[7]);
					zipsList.Remove(toks[1]);
				}
				
			}

		writeLatLon();
	}


    public static void writeLatLon()
    {

        StreamWriter output = new StreamWriter("LatLon.txt");

        int counter = 0;

        while (counter < latitude.Count())
        {

            output.Write(latitude[counter]);
            output.Write(" ");
            counter++;
            output.Write(latitude[counter]);
            output.Write("\n");
            counter++;

        }

        output.Close();
    } 


    public static void CityStates(){



		  
		StreamWriter output = new StreamWriter("CityStates.txt");
	 	List<String> tempList;
		tempList = citiesList.ConvertAll(d => d.ToUpper());
		citiesList = tempList;

		StreamReader input = new StreamReader("zipcodes.txt");
			
			while (!input.EndOfStream) {
				string line = input.ReadLine();
				string[] toks = line.Split('\t');
				if(citiesList.Contains(toks[3])){
				
					nvc.Add(toks[3], toks[4]);
					
				}
				
			}
	   string[] values = null;


           // need to change this to print keys in VALUES 
           foreach (string key in citiesList) {
		

		List<String> printList = new List<String>();
		
                values = nvc.GetValues(key);

                foreach (string value in values)
                {
                    
		   printList.Add(value);

                } 

		printList.Sort();
		List<String> noDuplicates = printList.Distinct().ToList();
		printList = noDuplicates;


	  

		foreach (string value in printList){
                    
		   output.Write(value);
		   output.Write(" ");

                } // end foreach
 	
		  output.Write("\n");
		

    } 
			
		output.Close();
} 

		
		public static void Main(string[] args){



                populateLists();
		CommonCityNames();
		LatLon();
		CityStates();
		
			
		}



} 

File: Zipcode Parsing/zipcodes.py
import sys
import re
import operator

global citiesList
global zipsList
global statesList

citiesList = []

global zipCodeList
zipCodeList = []

global upperCitiesList
upperCitiesList = []


global commonCityNames 
commonCityNames = []

global latitude
latitude = []

global citySt
citySt = dict()

global commonCityNamesDict
commonCityNamesDict = dict()


def read_text_file(fileName):
	global citiesList
	global zipsList
	global statesList

	with open (fileName) as file_txt:
	
		if fileName == 'cities.txt':
			citiesList = file_txt.read().splitlines()
		if fileName == 'zips.txt':
			zipsList = file_txt.read().splitlines()
		if fileName == 'states.txt':
			statesList = file_txt.read().splitlines()



def CommonCityNames():

	read_text_file('states.txt')
	readFile = open('zipcodes.txt', 'r')

	for line in readFile:
		pieces = line.split('\t')

		#commonCityNamesDict
		#if pieces[4] in statesList:
			#commonCityNames.append(pieces[3])
		
		if pieces[4] in statesList:
			key = pieces[4]

			if key not in commonCityNamesDict:
				commonCityNamesDict[key] = []

			commonCityNamesDict[key].append(pieces[3])

	writeFile()

def writeFile():

	
	commonCityNamesFinal = []
	writeFile = open('CommonCityNames.txt', 'w')

	# Find the intersection of all values in the commonCityNamesDictionary
	cityIntersection = set.intersection(*(set(val) for val in commonCityNamesDict.values()))

	# PrintCommonCityNames has all of the unique cities for each state in statesList.
	printCommonCityNames = list(cityIntersection)
	printCommonCityNames.sort()
		
	for item in printCommonCityNames:
		
		writeFile.write(item)
		writeFile.write('\n')


def writeFile2():
	writeFile = open('LatLon.txt', 'w')
	
	counter = 0
	
	while(counter < len(latitude)):

		writeFile.write(latitude[counter])
		writeFile.write(" ")
		counter = counter + 1
		writeFile.write(latitude[counter])
		counter = counter + 1
		writeFile.write('\n')


def LatLon():
	
	read_text_file('zips.txt')
	readFile = open('zipcodes.txt', 'r')
	
	#zipsList

	for line in readFile:
		
		pieces = line.split('\t')
		if pieces[1] in zipsList:
			latitude.append(pieces[6])
			latitude.append(pieces[7])
			zipsList.remove(pieces[1])

	writeFile2()

def writeFile3():

	writeFile = open('CityStates.txt', 'w')
	
	for cityName in citiesList:
	
		tempList = []
		tempList2 = []
		tempList = citySt[cityName.upper()]
		tempList2 = list(set(tempList))
		tempList2.sort()
	
		for elements in tempList2:
			writeFile.write(elements)
			writeFile.write(' ')


		writeFile.write('\n')
		

def CityStates():

	read_text_file('cities.txt')
	readFile = open('zipcodes.txt', 'r')
	upperCitiesList = [element.upper() for element in citiesList]


	for line in readFile:
		pieces = line.split('\t')

		if pieces[3] in upperCitiesList:
			key = pieces[3]

			if key not in citySt:
				citySt[key] = []

			citySt[key].append(pieces[4])

	writeFile3()
		

CommonCityNames()
LatLon()
CityStates()


